### ISSUE TITLE
Gittyup always crashes after 30 seconds

### ISSUE THREAD

**Issue by elman22** [Murmele/Gittyup#276]
When I start Gittyup, it always crashes after some random time. I have seen for example 30, 45 or 56 seconds. I don't need to do anything, just start the program with no repo is selected. I have also tried with clean install and same thing happens. Log doesn't say much:

```
❯ com.github.Murmele.Gittyup -d
Theme dir:  QDir( "/app/Resources/themes" , nameFilters = { "*" },  QDir::SortFlags( Name | IgnoreCase ) , QDir::Filters( Dirs|Files|Drives|AllEntries ) )
Theme dir:  QDir( "/app/Resources/themes" , nameFilters = { "*" },  QDir::SortFlags( Name | IgnoreCase ) , QDir::Filters( Dirs|Files|Drives|AllEntries ) )
Theme dir:  QDir( "/app/Resources/themes" , nameFilters = { "*" },  QDir::SortFlags( Name | IgnoreCase ) , QDir::Filters( Dirs|Files|Drives|AllEntries ) )
l10n dir:  QDir( "/app/Resources/l10n" , nameFilters = { "*" },  QDir::SortFlags( Name | IgnoreCase ) , QDir::Filters( Dirs|Files|Drives|AllEntries ) )

~ took 31s
```

I'm using Arch Linux + Flatpak

```
Name                          Application ID                            Version      Branch          Installation
Gittyup                       com.github.Murmele.Gittyup                v1.1.2       stable          system
```

Is there any way to enable more debug messages?

**Comment by Murmele** [comment]
normaly I run it as flatpak run com.github.Murmele.Gittyup

You should get more information then.
Is it just for one repository or for multiple. I'm using Manjaro (Arch based) and I don't see any issues. So we have to compare our setup

**Comment by elman22** [comment]
Hi. Thanks for advice. Running `flatpak run com.github.Murmele.Gittyup` didn't help. Gittyup still crashes and I get the same messages as before. Executable `com.github.Murmele.Gittyup` is flatpak export, so should work the same.

```
❯ which com.github.Murmele.Gittyup
/var/lib/flatpak/exports/bin/com.github.Murmele.Gittyup
```

I also tried AUR package, but there's something wrong with it because when launched, it searches for Lua API in my build folder which doesn't exist after the build.

**Comment by Murmele** [comment]
> 

Yes currently the AUR package is not working yet. There is a pullrequest pending.

**Comment by Murmele** [comment]
Which desktop environment are you using? Gnome, KDE, ...?

**Comment by elman22** [comment]
I have managed to fix PKGBUILD to install version 1.1.2 and so far it works fine.
I use KDE Plasma.

**Comment by Murmele** [comment]
Ah cool thanks. You can create a pullrequest then we can merge the changes here

**Comment by Murmele** [comment]
It is strange that you don't get that much debug output. When I open Gittyup and then a repository I get additional output

**Comment by elman22** [comment]
It crashes even with no repo opened. If I open repo, I see more debug output, but I didn't think it was relevant.

However I tried Gittyup on my second computer which has Manjaro and it's working without any problems. So must be something with Arch or my Arch config.

**Comment by Murmele** [comment]
Maybe a qt issue?

**Comment by elman22** [comment]
I'm using Manjaro Unstable, so I have exactly the same Qt versions. Difference is that Manjaro uses X11 and Arch uses Wayland.

Anyway I just stumbled upon message in journalctl:

```
xdg-desktop-portal-kde[1235]: This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
```

This may be the problem since I assume that flatpak uses portal and AUR package not.

**Comment by Murmele** [comment]
Can you try to set the Qt platform plugin?

**Comment by elman22** [comment]
Hi. I have installed AUR version of 1.2.2 and so far I haven't seen any crash. So I hope the issue is resolved. Thanks a lot :smiley:
